### PR TITLE
Update prometheus-adapter module to latest chart and Docker versions

### PR DESCRIPTION
**Pull request by tunguyen9889**
- Update Chart and Docker image to latest versions.
- Update values.yaml to match with latest Chart version.
- Add CODEOWNERS and update linter image.

### CODE REVIEW

File: templates/values.yaml [hunk 1]
```diff
@@ -151,9 +156,32 @@ extraVolumeMounts: ${extra_volume_mounts}
...
 
+hostNetwork:
+  # Specifies if prometheus-adapter should be started in hostNetwork mode.
+  #
+  # You would require this enabled if you use alternate overlay networking for pods and
+  # API server unable to communicate with metrics-server. As an example, this is required
+  # if you use Weave network on EKS. See also dnsPolicy
+  enabled: ${host_network_enabled}
+
+%{ if host_network_enabled == true ~}
```
Comment (lawliet89) [inline templates/values.yaml:173]:
```suggestion
%{ if host_network_enabled ~}
```

File: variables.tf [hunk 2]
```diff
@@ -73,6 +73,11 @@ variable "resources" {
   default     = {}
 }
 
+variable "host_network_enabled" {
+  description = "Specifies if prometheus-adapter should be started in hostNetwork mode."
+  default     = false
```
Comment (lawliet89) [inline variables.tf:78]: Add `type = bool`

File: versions.tf [hunk 3]
```diff
@@ -0,0 +1,10 @@
+terraform {
+  required_version = ">= 0.14"
+
+  required_providers {
+    helm = {
+      source  = "hashicorp/helm"
+      version = "~> 2.1"
```
Comment (lawliet89) [inline versions.tf:7]:
```suggestion
      version = ">= 2.1"
```

Use less restrictive version constraints in modules.

**Review by lawliet89 — CHANGES_REQUESTED** [review]

File: variables.tf (hunk 2 above)
Comment (tunguyen9889) [reply variables.tf:79]: Fixed

File: templates/values.yaml (hunk 1 above)
Comment (tunguyen9889) [reply templates/values.yaml]: fixed

File: versions.tf (hunk 3 above)
Comment (tunguyen9889) [reply versions.tf]: fixed

**Review by lawliet89 — APPROVED** [review]

**Pull request merged**